From: Simon McVittie <smcv@collabora.com>
Date: Tue, 13 Feb 2024 10:17:53 +0000
Subject: gdatetime: Fix ja_JA.eucjp expectations when eras are unsupported
MIME-Version: 1.0
Content-Type: text/plain; charset="utf-8"
Content-Transfer-Encoding: 8bit

I can't read Japanese, but these match the output of:

    env TZ=UTC LC_ALL=ja_JP.eucjp date -d2009-10-24T00:00:00Z '+%c' | iconv -f eucjp -t UTF-8
    env TZ=UTC LC_ALL=ja_JP.eucjp date -d2009-10-24T00:00:00Z '+%C' | iconv -f eucjp -t UTF-8

which seem like a reasonable thing to use as a reference.

According to Google Translate, 平成 refers to the Heisei era, which was
current during 2009, and seems unreasonable to expect as output on a
platform where era-based dates are unsupported.

Bug: https://gitlab.gnome.org/GNOME/glib/-/issues/3252
Signed-off-by: Simon McVittie <smcv@collabora.com>
Forwarded: https://gitlab.gnome.org/GNOME/glib/-/merge_requests/3920
---
 glib/tests/gdatetime.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/glib/tests/gdatetime.c b/glib/tests/gdatetime.c
index c6a6fed..f2de1ab 100644
--- a/glib/tests/gdatetime.c
+++ b/glib/tests/gdatetime.c
@@ -1733,8 +1733,8 @@ test_non_utf8_printf (void)
   TEST_PRINTF ("%Ey", "21");
   TEST_PRINTF ("%EY", "平成21年");
 #else
-  TEST_PRINTF ("%Ec", "平成21年10月24日 00時00分00秒");
-  TEST_PRINTF ("%EC", "平成");
+  TEST_PRINTF ("%Ec", "2009年10月24日 00時00分00秒");
+  TEST_PRINTF ("%EC", "20");
   TEST_PRINTF ("%Ex", "2009\345\271\26410\346\234\21024\346\227\245");
   TEST_PRINTF ("%EX", "00\346\231\20200\345\210\20600\347\247\222");
   TEST_PRINTF ("%Ey", "09");
